internal/store/memory/store.go

```diff
@@ -219,11 +219,7 @@ func (s *Store) UpsertGateway(ctx context.Context, gateway store.Gateway) error
 
 	current, found := s.gateways[id]
 
-	switch current.Compare(gateway) {
-	case store.CompareResultInvalid, store.CompareResultNewer:
-		// we have an invalid or old route, ignore it
-		return nil
-	case store.CompareResultNotEqual:
+	if current.ShouldUpdate(gateway) {
```

Comment:
Think we need to vary this logic a slight amount -- after this block there's still a call to `sync` gateways. Though I'm pretty sure the `sync` calls will just no-op since nothing has been updated, to be in accordance with the  previous logic we should do something like:

```go
if !current.ShouldUpdate(gateway) {
  // no-op
  return nil
}
```